src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -55,12 +55,22 @@ def __init__(self, **kwargs):
         self.cloud = get_active_cloud(self)
         logger.debug('Current cloud config:\n%s', str(self.cloud.name))
 
+        self._patch_extensions_dir()
+
         register_global_transforms(self)
         register_global_subscription_argument(self)
         register_ids_argument(self)  # global subscription must be registered first!
 
         self.progress_controller = None
 
+    def _patch_extensions_dir(self):
+        from importlib import import_module
+        from azure.cli.core._config import GLOBAL_CONFIG_DIR
+        ext_module = import_module('azure.cli.core.extension')
+        custom_ext_dir = self.config.get('extension', 'dir', None)
+        setattr(ext_module, 'EXTENSIONS_DIR', os.path.expanduser(custom_ext_dir)
+                if custom_ext_dir else os.path.join(GLOBAL_CONFIG_DIR, 'cliextensions'))
+
```

Comment:
Definitely looks very hacky. Is there no other way to get at the config without having a CLI context?

Reply:
There's no way that's less of a hack.

Reply:
So from a first glance, parsing the config before setting the global variable [here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/extension/__init__.py#L16) already seems like a better option.
I would think modifying global variables from other modules during runtime is something we should try to avoid.

Reply:
https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/_config.py should have all the information needed to locate the config. Please consider reading from the file instead of the current logic.